automation/gtm_automation.js

```diff
@@ -0,0 +1,496 @@
+    client_id: clientId,
+    scope: 'https://www.googleapis.com/auth/tagmanager.edit.containers',
+    callback: '',
+  });
+  gapi.load('client', () => {
+    gapi.client.init({})
+      .then(function() {
+        gapi.client.load('https://tagmanager.googleapis.com/$discovery/rest?version=v2');
+      })
+      .then(runDeployment());
+  });
+}
+
+/**
+ * Start point for deploying the template, tags, etc. to GTM.
+ *
+ * The configuration is collected from the web interface here to centralize
+ * where the document is accessed. This config is then passed down the chain as
+ * the GTM API calls are made.
+ *
+ * The template file is read in here and then passed to createTemplate() along
+ * with the config to start the deployment workflow.
+ */
+function runDeployment() {
+  lockForm();
+  // set up the global config
+  const gtmURL = document.getElementById('gtm-url').value;
+  const config = new Map();
+  config.set('parent', gtmURL.substring(gtmURL.indexOf('accounts')));
+  config.set('ga4ID', document.getElementById('ga4-id').value);
+  config.set('metrics', new Map(Object.entries({
+    ttfb: document.getElementById('cwv-TTFB').checked.toString(),
+    fcp: document.getElementById('cwv-FCP').checked.toString(),
+    lcp: document.getElementById('cwv-LCP').checked.toString(),
+    cls: document.getElementById('cwv-CLS').checked.toString(),
+    fid: document.getElementById('cwv-FID').checked.toString(),
+    inp: document.getElementById('cwv-INP').checked.toString(),
+  })));
+  config.set('useUnpkg', document.getElementById('use-unpkg').checked);
+  config.set('libraryUrl', document.getElementById('library-url').value);
+  config.set('attributionBuild', document.getElementById('use-attribution').checked);
+  config.set('gtmEventName', document.getElementById('event-name').value);
+
+  // get the template content from the user. The file chooser is a mandiatory
+  // field.
+  const templateFile = document.getElementById('template-file').files[0];
+  const templateReader = new FileReader();
+  templateReader.addEventListener('load', () => {
+    createTemplate(config, templateReader.result)
+  });
+  templateReader.readAsText(templateFile);
+}
+
+/**
+ * Creates the template used to measure the Core web Vital Metrics in the
+ * provided workspace.
+ *
+ * If the template is successfully created, the config and the new template ID
+ * is passed to createTag() to continue the deployment.
+ *
+ * @param config {Map<string, *>} The configuration map.
+ * @param content {string} The content of the template.
+ */
+function createTemplate(config, content) {
+  tokenClient.callback = (resp) => {
+    if (resp.error !== undefined) {
+      throw (resp);
+    }
+    gapi.client.tagmanager.accounts.containers.workspaces.templates.create({
+      parent: config.get('parent'),
+      templateData: content,
+    })
+      .then((gtmResp) => {
+        addSuccessMessage('Created Web Vitals Template');
+        const templateID = gtmResp.result.templateId;
```

Comment:
nit: Should be templateId
https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
        const templateId = gtmResp.result.templateId;
```